src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -608,6 +597,55 @@ export class MongoStorageAdapter implements StorageAdapter {
       .catch(err => this.handleError(err));
   }
 
+  _parseAggregateArgs(schema: any, pipeline: any): any {
+    var rtnval = {};
+    if (Array.isArray(pipeline)) {
+      rtnval = [];
+      for (const field in pipeline) {
+        rtnval.push(this._parseAggregateArgs(schema, pipeline[field]));
+      }
+    }
+    else if (typeof pipeline === 'object') {
+      for (const field in pipeline) {
+        if (schema.fields[field] && schema.fields[field].type === 'Pointer') {
+          rtnval[`_p_${field}`] = `${schema.fields[field].targetClass}$${pipeline[field]}`;
+        }
+        else if (schema.fields[field] && schema.fields[field].type === 'Date') {
+          rtnval[field] = this._convertToDate(pipeline[field]);
+        }
+        else {
+          rtnval[field] = this._parseAggregateArgs(schema, pipeline[field]);
+        }
+
+        if (field === 'objectId') {
+          rtnval['_id'] = rtnval[field];
+          delete rtnval[field];
+        }
+        else if (field === 'createdAt') {
+          rtnval['_created_at'] = rtnval[field];
+          delete rtnval[field];
+        }
+        else if (field === 'updatedAt') {
+          rtnval['_updated_at'] = rtnval[field];
+          delete rtnval[field];
+        }
+      }
+    }
+    return rtnval;
+  }
+
+  _convertToDate(value: any): any {
+    if (typeof value === 'string') {
+      return new Date(value);
+    }
+
+    var rtnval = {}
+    for (const field in value) {
```

Comment:
can we document what we expect the value to look like, I have trouble understanding what it does, and why we need the iterator, and recursion.

Reply:
Sorry for the newbie question...how do we take care of these? I've updated my branch to include your suggestions. Thanks.